dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -693,7 +693,18 @@ internal ClientCore(ILogger? logger = null)
                             OpenAIFunctionToolCall.TrackStreamingToolingUpdate(update.ToolCallUpdate, ref toolCallIdsByIndex, ref functionNamesByIndex, ref functionArgumentBuildersByIndex);
                         }
 
-                        var openAIStreamingChatMessageContent = new OpenAIStreamingChatMessageContent(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, metadata) { AuthorName = streamedName };
+                        AuthorRole? role = null;
+                        if (streamedRole.HasValue)
+                        {
+                            role = new AuthorRole(streamedRole.Value.ToString());
+                        }
+
+                        OpenAIStreamingChatMessageContent openAIStreamingChatMessageContent =
+                            new(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, metadata)
+                            {
+                                AuthorName = streamedName,
+                                Role = role,
+                            };
```

Comment:
nit: I would use ternary operator here for simplicity, since this method is already big enough, but feel free to proceed with version that is more comfortable for you:
```suggestion
                        OpenAIStreamingChatMessageContent openAIStreamingChatMessageContent =
                            new(update, update.ChoiceIndex ?? 0, this.DeploymentOrModelName, metadata)
                            {
                                AuthorName = streamedName,
                                Role = streamedRole.HasValue ? new AuthorRole(streamedRole.Value.ToString()) : null
                            };
```